### ISSUE TITLE
Sample should have private endpoint for Key Vault and Storage Account

### ISSUE THREAD

**Issue by francovanzyl96** [Azure/azure-functions-on-container-apps#93]
**Is your feature request related to a problem? Please describe.**
The sample would be more practical and secure if the Azure Key Vault and Storage Account resources have private endpoints deployed. Additionally the Azure Container Registry could also have a private endpoint.

**Describe the solution you'd like**
I would like to see a sample where Azure Key Vault and Storage Account resources have private endpoints deployed. Additionally the Azure Container Registry could also have a private endpoint. I would also like to see samples of how the App Settings should be configured to force all outbound traffic from the Function App to go over the private VNet attached to the Container Apps Environment.

**Describe alternatives you've considered**
I have attempted to use the App Settings as described for deploying Function Apps on App Service Plans, however this does not seem to include a solution for accessing Key Vault secrets over a private endpoint connection.

**Additional context**

**Comment by raorugan** [comment]
Ack on this. Allow us to get back on this.

**Comment by raorugan** [comment]
@francovanzyl96 , I was able to set storage app setting with private endpoint for a function app deployed on ACA , here is the sample endpoint for the same

![Image](https://github.com/user-attachments/assets/4d00b9e8-98cd-4c79-a4da-d0df1049a286)

Can you clarify following:
1. Which SKU - app service or Elastic premium or azure container apps is the function app deployed?
2. What error message do you see?

**Comment by raorugan** [comment]
We need resource id details for this to debug further. Since this is a public forum can you please open support ticket ? We can get your resource details and check why provisioning failed?

On a side note did you try re-deploying , ensure resource provisioning is successful (you may check this in the json view of the resource in the Azure Portal) and trigger the durable function?

**Comment by raorugan** [comment]
Hi @francovanzyl96 , were you able to raise support issue on this. On the other hand can you check if  private DNS or private endpoint or IP config issues exist. As a repro suggested the same error message that you were seeing due to network config issues.

**Comment by francovanzyl96** [comment]
Hi @raorugan, I was not able to raise a support ticket on this yet, would need to redeploy an environment with private endpoints to reproduce the issue again first. Will do this asap and create a support request.

On the other hand can you check if private DNS or private endpoint or IP config issues exist - I have checked that private endpoints and private DNS zones are properly created and linked to the correct VNets. Are there any specific checks you would like me to do?

**Comment by francovanzyl96** [comment]
Ok thanks and also Key Vault?

**Comment by raorugan** [comment]
yes , KeyVault as well. Azure service that is reachable through private endpoint from container app environment VNET Internal with managed identity should work

**Comment by francovanzyl96** [comment]
Ok, I will redeploy it like that and create a support ticket. Thanks for looking into it.
